src/Bindings/ManualBindings.cpp

```diff
@@ -2775,6 +2776,57 @@ static int tolua_cCompositeChat_UnderlineUrls(lua_State * tolua_S)
 
 
 
+static int tolua_cChannelManager_RegisterChannel(lua_State * tolua_S)
+{
+	
+	// Retrieve the cPlugin from the LuaState:
+	cPluginLua * Plugin = cManualBindings::GetLuaPlugin(tolua_S);
+	if (Plugin == nullptr)
+	{
+		// An error message has been already printed in GetLuaPlugin()
+		return 0;
+	}
+	
+	// Check params:
+	cLuaState L(tolua_S);
+	if (!L.CheckParamUserType(1, "cChannelManager"))
+	{
+		return 0;
+	}
+	cChannelManager * self = reinterpret_cast<cChannelManager *>(tolua_tousertype(tolua_S, 1, nullptr));
+	if (self == nullptr)
+	{
+		return cManualBindings::lua_do_error(tolua_S, "invalid 'self' in function 'cCompositeChat:UnderlineUrls'");
+	}
+	if (!L.CheckParamString(2))
+	{
+		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Expected a string for parameter #1");
+	}
+	if (!L.CheckParamFunction(3))
+	{
+		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Expected a function for parameter #2");
+	}
```

Comment:
The `CheckParamXYZ()` functions already do the error output. Have a look at the other bindings for how to use these. Usually one big `if` statement is enough.